### PR TITLE
[1.21.4] Sync remaining `c` tags from Fabric side

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
Neoforge 1.21.1 port: https://github.com/neoforged/NeoForge/pull/1593
Fabric 1.21.4 PR: https://github.com/FabricMC/fabric/pull/4186

Note, I did not add any new Pale Garden stuff myself to the tags because those will need special handling to only add to the tags when the user activates the feature flag. Maybe that is better split into a separate PR by someone with more experience with feature flags than me.

---------------
Fabric had some tags that was not in Neo. This PR aims to solve that by importing those tags. Specifically `c:potions` item tag and `c:is_temperate` biome tags. I added the torchflower seeds to `c:seeds/torchflower` and then added that to the `c:seeds` tag as well as it was a missing seed. 

I also improved javadoc for several tags, did some cleanup in a few spots, and marked the `c:is_modified` biome tag as deprecated and for removal in 1.22. Reason being the modified tag was based off of how the old biome layout used to work in the long past. Since Mojang revamped how biomes work, this tag lost all of its meaning and serves no purpose anymore. Hence time to retire it.

The PR to add the tags that are not in Fabric from Neoforge is in this other PR. Discussions ongoing of course: https://github.com/FabricMC/fabric/pull/4186

The Nether stem and hyphae was missing from the Stripped wood/logs tag. This is fixed as well.

### CODE REVIEW

File: src/generated/resources/assets/c/lang/en_us.json [hunk 1]
```diff
@@ -164,6 +166,8 @@
   "tag.item.c.chests.ender": "Ender Chests",
   "tag.item.c.chests.trapped": "Trapped Chests",
   "tag.item.c.chests.wooden": "Wooden Chests",
+  "tag.item.c.clumps": "Clumps",
+  "tag.item.c.clumps.resin": "Clumps Resin",
```
Comment (ageuxo) [inline src/generated/resources/assets/c/lang/en_us.json]: shouldn't this be "Resin Clumps" or "Clumps of Resin"?

File: src/generated/resources/assets/c/lang/en_us.json [hunk 2]
```diff
@@ -302,6 +306,8 @@
   "tag.item.c.ores_in_ground.stone": "Stone Ores In Ground",
   "tag.item.c.player_workstations.crafting_tables": "Crafting Tables",
   "tag.item.c.player_workstations.furnaces": "Furnaces",
+  "tag.item.c.potions": "Potions",
+  "tag.item.c.potions.bottle": "Bottle Potions",
```
Comment (ageuxo) [inline src/generated/resources/assets/c/lang/en_us.json]: This should maybe be "Bottled Potions"?

File: src/main/java/net/neoforged/neoforge/common/Tags.java [hunk 3]
```diff
@@ -572,6 +585,18 @@ public static class Items {
...
+         * Contents of this tag may not always be a kind of bottle. Buckets of potions could go here.
+         * The subtags would be the name of the container that is holding the potion effects such as `c:potions/bucket` or `c:potions/vial` as examples.
+         */
+        public static final TagKey<Item> POTIONS = tag("potions");
+        /**
+         * Variations of the potion bottle that can hold various effects by using {@link net.minecraft.core.component.DataComponents#POTION_CONTENTS}.
+         * Examples are splash and lingering potions from vanilla.
+         * If a mod adds a new variant like a seeking potion that applies effect to the closest entity at impact, that would in this tag.
+         */
+        public static final TagKey<Item> POTION_BOTTLE = tag("potions/bottle");
```
Comment (dhyces) [inline src/main/java/net/neoforged/neoforge/common/Tags.java]: Seems like other tags retain the "parent" tag name in their names, ie `CLUMPS_RESIN` or `BRICKS_RESIN`. I think keeping that pattern here makes sense.
```suggestion
        public static final TagKey<Item> POTIONS_BOTTLE = tag("potions/bottle");
```

**Review by dhyces — CHANGES_REQUESTED** [review]
- Noticed that the Fabric PR used a different translation for "tag.block.c.fences.nether_brick". Neo's is currently "Fences Nether Brick" and the Fabric PR (as well as Neo's item tag) uses "Nether Brick Fences".
- Fabric PR uses singular "Plateau" while Neo is using plural "Plateaus" for "tag.worldgen.biome.c.is_plateau"

**Review by dhyces — DISMISSED** [review]

**Comment by TelepathicGrunt** [comment]
Fabric side merged. Will get this merged now

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**